app/models/declaration.rb

```diff
@@ -172,9 +181,14 @@ def duplicate_declaration_exists?
 
     existing_declarations
       .billable_or_changeable
-      .where(declaration_type:)
+      .where(declaration_type:, superseded_by_id: nil)
       .excluding(self)
-      .exists?
+  end
+
+  def duplicate_declaration_exists?
+    return unless billable_or_changeable?
```

Comment:
Does it need to be `billable_or_changeable?`? I had a quick look at ECF and I don't think it makes this distinction?